pkg/security/ebpf/c/defs.h

```diff
@@ -3,16 +3,136 @@
 
 #include "../../../ebpf/c/bpf_helpers.h"
 
+#if defined(__x86_64__)
+  #define SYSCALL64_PREFIX "__x64_"
+  #define SYSCALL32_PREFIX "__ia32_"
+
+  #define SYSCALL64_PT_REGS_PARM1(x) ((x)->di)
+  #define SYSCALL64_PT_REGS_PARM2(x) ((x)->si)
+  #define SYSCALL64_PT_REGS_PARM3(x) ((x)->dx)
+  #define SYSCALL64_PT_REGS_PARM4(x) ((x)->r10)
```

Comment:
Why is this `r10`? I see it as `cx` in other sources.

Reply:
cx is for the C convention. x64 syscalls ABI uses r10 instead (https://github.com/torvalds/linux/blob/master/arch/x86/entry/entry_64.S#L83 and https://en.wikibooks.org/wiki/X86_Assembly/Interfacing_with_Linux)

Reply:
Something to be aware of. When you use the `__x64_` prefixed syscall, you can only access the arguments via indirection. See https://github.com/iovisor/bcc/commit/2da34267fcae4485f4e05a17521214749f6f0edd and https://github.com/torvalds/linux/commit/fa697140f9a20119a9ec8fd7460cc4314fbdaff3 for context.

You may already be handling this. I wasn't able to follow your macros.

Reply:
We indeed take care of this indirection. The previous version was easier to read :-)